catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/fs/FileSystemUtils.java

```diff
@@ -0,0 +1,96 @@
+import com.google.common.collect.Streams;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FileSystemUtils {
+
+  private FileSystemUtils() {}
+
+  public static Map<String, FileSystemProvider> getFileSystemProviders(String fileSystemProviders) {
+    Map<String, FileSystemProvider> resultMap = Maps.newHashMap();
+    ServiceLoader<FileSystemProvider> allFileSystemProviders =
+        ServiceLoader.load(FileSystemProvider.class);
+
+    Set<String> providersInUses =
+        fileSystemProviders != null
+            ? Arrays.stream(fileSystemProviders.split(","))
+                .map(String::trim)
+                .collect(java.util.stream.Collectors.toSet())
+            : Sets.newHashSet();
+
+    // Add built-in file system providers to the use list automatically.
+    providersInUses.add(BUILTIN_LOCAL_FS_PROVIDER);
+    providersInUses.add(BUILTIN_HDFS_FS_PROVIDER);
+
+    // Only get the file system providers that are in the user list and check if the scheme is
+    // unique.
+    Streams.stream(allFileSystemProviders.iterator())
+        .filter(fileSystemProvider -> providersInUses.contains(fileSystemProvider.name()))
+        .forEach(
+            fileSystemProvider -> {
+              if (resultMap.containsKey(fileSystemProvider.scheme())) {
+                throw new UnsupportedOperationException(
+                    String.format(
+                        "File system provider: '%s' with scheme '%s' already exists in the use provider list "
+                            + "Please make sure the file system provider scheme is unique.",
+                        fileSystemProvider.getClass().getName(), fileSystemProvider.scheme()));
+              }
+              resultMap.put(fileSystemProvider.scheme(), fileSystemProvider);
+            });
+
+    // If not all file system providers in providersInUses was found, throw an exception.
+    Set<String> notFoundProviders =
+        Sets.difference(
+                providersInUses,
+                resultMap.values().stream()
+                    .map(FileSystemProvider::name)
+                    .collect(Collectors.toSet()))
+            .immutableCopy();
+    if (!notFoundProviders.isEmpty()) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "File system providers %s not found in the classpath. Please make sure the file system "
+                  + "provider is in the classpath.",
+              notFoundProviders));
+    }
+
+    return resultMap;
+  }
+
+  public static FileSystemProvider getFileSystemProviderByName(
+      Map<String, FileSystemProvider> fileSystemProviders, String fileSystemProviderName) {
+    return fileSystemProviders.entrySet().stream()
+        .filter(entry -> entry.getValue().name().equals(fileSystemProviderName))
+        .map(Map.Entry::getValue)
+        .findFirst()
+        .orElse(null);
```

Comment:
What if it returns null here, I don't see you handle this "null" case.

Reply:
Sorry, forgot to change here accordingly.